protocols/virtualdefund/virtualdefund.go

```diff
@@ -217,6 +248,71 @@ func (o Objective) fullySigned() bool {
 	return true
 }
 
+// isAlice returns true if the receiver represents participant 0 in the virtualdefund protocol
+func (o Objective) isAlice() bool {
+	return o.MyRole == 0
+}
+
+// isBob returns true if the receiver represents participant 2 in the virtualdefund protocol
+func (o Objective) isBob() bool {
+	return o.MyRole == 2
+}
+
+// ledgerProposal generates a ledger proposal to remove the guarantee for V for ledger
+func (o Objective) ledgerProposal(ledger *consensus_channel.ConsensusChannel) consensus_channel.Proposal {
+	left := o.finalOutcome().Allocations[0].Amount
+	right := o.finalOutcome().Allocations[1].Amount
+
+	return consensus_channel.NewRemoveProposal(ledger.Id, FinalTurnNum, o.VId(), left, right)
+}
+
+// defundLedger updates the ledger channel to remove the guarantee that funds V.
+func (o *Objective) defundLedger(ledger *consensus_channel.ConsensusChannel, sk *[]byte) (protocols.SideEffects, error) {
```

Comment:
Nit: this sounds like imperative code again. Would it be better to name this (or at least describe this in the godoc) as "attempting to defund the ledger by generating side effects "?

Reply:
Also, the name is ambiguous. We aren't trying to defund the ledger here, we are trying to defund the virtual channel.

Something like `generateProposalToRemoveFromLedger`?

Reply:
Nit: this sounds like imperative code again. Would it be better to name this (or at least describe this in the godoc) as "attempting to defund the ledger by generating side effects "?

This function modifies the `ledger` consensus channel by adding the proposal to the channel, so I think naming it `generateProposalToRemoveFromLedger` could be slightly misleading.

I've renamed this to `updateLedgerToRemoveGuarantee` which mirrors `updateLedgerWithGuarantee` in virtual funding.

Reply:
Right: I think what is throwing me is that it sounds like we are unilaterally updating the ledger. When all we are doing is updating our side and generating a message to send. But maybe it's too fussy to try and reflect that in the name! 

`updateLedgerToRemoveFromLedger` is definitely better than `defundLedger`.